### PR TITLE
Readme update + StyleCop

### PR DESCRIPTION
**Pull request by scowalt**
Two biggest changes here:

1. README has been updated to include some fancy styling and badges. Here, my goal was to match the trends of active popular open-source repositories.
1. Integrate StyleCop into the repository. For the most part, I disabled all of the rules that would have a significant impact on the style of the code. I made two exceptions to this:
    1.  I updated `using` directives to be inside of a namespace. See [SA1200](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1200.md)
    1. I removed many, many tab characters from the code and replaced them with a set of four spaces each

I made some other minor changes. The most noteworthy:

- Some `.csproj` file updates to remove build warnings on my end. Please look over these modifications as I have very little insight into the broader impact of my changes.
- I updated the license file to extend the copyright year.
- CI integration will now run on pushes *and* pull requests
- Many trailing whitespace were removed

### CODE REVIEW

File: Source/Libraries/CorruptCore/CorruptCore.csproj [hunk 1]
```diff
@@ -44,7 +44,6 @@
     <PlatformTarget>x64</PlatformTarget>
     <LangVersion>latest</LangVersion>
     <ErrorReport>prompt</ErrorReport>
-    <CodeAnalysisRuleSet>MinimumRecommendedRules.ruleset</CodeAnalysisRuleSet>
```
Comment (scowalt) [inline Source/Libraries/CorruptCore/CorruptCore.csproj:47]: I need to check if I need to add the new ruleset to this file.

Comment (scowalt) [reply Source/Libraries/CorruptCore/CorruptCore.csproj:47]: It looks like the rulesets from elsewhere propagate to this `csproj` so I shouldn't need to make anymore changes here.

File: Source/Libraries/PluginHost/PluginHost.csproj [hunk 2]
```diff
@@ -34,7 +34,7 @@
     <Reference Include="System" />
     <Reference Include="System.Configuration" />
     <Reference Include="System.Core" />
-    <Reference Include="System.IO.Compression, Version=4.2.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089, processorArchitecture=MSIL" />
```
Comment (scowalt) [inline Source/Libraries/PluginHost/PluginHost.csproj:37]: This is a change I had to make to remove a build warning, but I'm not sure of its broader implications.

File: StyleCop.ruleset [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+    <Rule Id="CA2235" Action="Warning" />
+    <Rule Id="CA2236" Action="Warning" />
+    <Rule Id="CA2237" Action="Warning" />
+    <Rule Id="CA2238" Action="Warning" />
+    <Rule Id="CA2240" Action="Warning" />
+    <Rule Id="CA2241" Action="Warning" />
+    <Rule Id="CA2242" Action="Warning" />
+    <Rule Id="CS0649" Action="None" /> <!-- Field is never assigned to -->
+  </Rules>
+</RuleSet>
```
Comment (scowalt) [inline StyleCop.ruleset:149]: Addd trailing newline

**Comment by scowalt** [comment]
I highly recommend a "squash and merge" when integrating this to avoid my messy commit history.

File: README.md [hunk 4]
```diff
@@ -22,11 +42,10 @@ Features:
 - The Vector Engine is a specialty corruption engine designed for 3D systems. The included lists target systems that support IEEE754 Floating Point values
 - The Custom Engine allows you to create your own engine using the various corruption parameters.
 
+## Development
```
[the remark this answers is not in the document]
Comment (NarryG) [reply README.md:45]: I'm fine with having the section, but I'd like it to mention VS2019 as the primary method if we're adding other methods

File: Source/Libraries/PluginHost/PluginHost.csproj (hunk 2 above)
Comment (NarryG) [reply Source/Libraries/PluginHost/PluginHost.csproj:37]: I don't think this should have even been referenced in there in the first place, but I'll double-check

**Comment by NarryG** [comment]
I forsee some potential issues with this. The main one being that all the whitespace changes are going to nuke compatibility with the 505 dev branch and we'll never be able to cleanly merge it in. Could you rebase this onto the 505 dev branch?

**Comment by scowalt** [comment]
Yeah for sure. Do you want me to change the merge target too, or just rebase?

**Comment by NarryG** [comment]
> Yeah for sure. Do you want me to change the PR target too, or just rebase?

Rebase and change the target. Shouldn't cause too many conflicts with stuff people are working on since we commit regularly enough and there's no big functionality being worked on in its own branch right now

**Comment by NarryG** [comment]
Thanks again. This kind of cleanup was planned as part of the refactor, but it removes a step from that.  The code style isn't the most consistent thing (we never established a true style guide as it was originally just ircluzar then I hopped in).  

I'd be fine with migrating the code to conform to the C# standard more over time, (variable naming is all over the place as you probably saw), but that's a long project and for some aspects of the code (serialization), it'll be a breaking change.

**Comment by scowalt** [comment]
I'm working on a 505 branch rebase right now. I make might a separate PR to bring the `README.md` goodness into master because it looks nice 😄 

Yeah, I thought, after this PR, I might iterate more overtime to try to get this project closer to some set of standards, where it makes sense. I definitely don't want to break changes that may merge from elsewhere though. Hopefully, the 505 branch will be a good place for me to continue to bring in this work.

**Comment by scowalt** [comment]
Rebased and moved this to the 505 branch (and a new PR)

**Pull request closed without merging**